torch/fx/graph_module.py

```diff
@@ -35,14 +36,27 @@ def patched_getline(*args, **kwargs):
     return _orig_getlines(*args, **kwargs)
 linecache.getlines = patched_getline
 
-def _forward_from_src(src : str):
+
+def _forward_from_src(src: str, importer: Optional[PackageImporter]):
     # If you add more globals here, remember to add their names to fx.graph._shadows_builtin_name!
     gbls: Dict[str, Any] = {'inf': math.inf, 'nan': math.nan, 'NoneType' : type(None)}
+    if importer is not None:
+        gbls['__loader__'] = importer
+        gbls['__builtins__'] = importer.patched_builtins
+
     exec_with_source(src, gbls)
     return gbls['forward']
 
 
-def deserialize_graphmodule(body : dict) -> torch.nn.Module:
+def reduce_graph_module(body: dict) -> torch.nn.Module:
+    return _deserialize_graph_module(None, body)
+
+
+def reduce_package_graph_module(importer: PackageImporter, body: dict) -> torch.nn.Module:
+    return _deserialize_graph_module(importer, body)
+
+
+def _deserialize_graph_module(importer: Optional[PackageImporter], body: dict) -> torch.nn.Module:
```

Comment:
This funnels two paths through the same function where the very first line is an if-statement that splits the paths again. Makes more sense to put the code that loads forward into these functions and have `_deserialize_..` take it as an argument.